spec/lib/appsignal/garbage_collection_profiler_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'spec_helper'
+require_relative '../../support/mocks/fake_gc_profiler'
+
+describe Appsignal::GarbageCollectionProfiler do
+  before do
+    @internal_profiler = FakeGCProfiler.new
```

Comment:
I think it's better if you use `let`s here instead of instance variables unless you really need values to persist between specs.